### PR TITLE
Implement ITextInputMethodClient for TextArea

### PR DESCRIPTION
**Pull request by Gillibald**

### CODE REVIEW

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 1]
```diff
@@ -745,12 +756,14 @@ protected override void OnGotFocus(GotFocusEventArgs e)
             // First activate IME, then show caret
```
Comment (danipen) [inline src/AvaloniaEdit/Editing/TextArea.cs:756]: Does the order matters? Old comments said:
First activate IME, then show caret

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 2]
```diff
@@ -745,12 +756,14 @@ protected override void OnGotFocus(GotFocusEventArgs e)
...
             //ime.OnGotKeyboardFocus(e);
```
Comment (danipen) [inline src/AvaloniaEdit/Editing/TextArea.cs:757]: We can delete old commented code.

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 3]
```diff
@@ -745,12 +756,14 @@ protected override void OnGotFocus(GotFocusEventArgs e)
...
             Caret.Show();
+            _imClient.SetTextArea(this);
         }
 
         protected override void OnLostFocus(RoutedEventArgs e)
         {
             base.OnLostFocus(e);
             Caret.Hide();
+            _imClient.SetTextArea(null);
             //ime.OnLostKeyboardFocus(e);
```
Comment (danipen) [inline src/AvaloniaEdit/Editing/TextArea.cs:767]: We can delete old commented code.

**Review by danipen — CHANGES_REQUESTED** [review]

File: src/AvaloniaEdit/Editing/TextArea.cs (hunk 1 above)
Comment (Gillibald) [reply src/AvaloniaEdit/Editing/TextArea.cs:756]: This is handled by Avalonia so a control that wants to interact with IME just has to implement `ITextInputMethodClient`

Comment (Gillibald) [reply src/AvaloniaEdit/Editing/TextArea.cs:756]: Will change order just for consistency

**Review by danipen — APPROVED** [review]

**Pull request merged**